File: app/src/main/java/com/ternaryop/tumblr/TumblrFollowers.java
package com.ternaryop.tumblr;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by dave on 18/01/15.
 * Info about followers
 */
public class TumblrFollowers implements Serializable {
    private long totalUsers;
    private ArrayList<TumblrUser> usersList;

    public TumblrFollowers() {
        usersList = new ArrayList<>();
    }

    public void add(JSONObject json) throws JSONException {
        totalUsers = json.getLong("total_users");
        JSONArray users = json.getJSONArray("users");
        for (int i = 0; i < users.length(); i++) {
            usersList.add(new TumblrUser(users.getJSONObject(i)));
        }
    }

    public List<TumblrUser> getUsersList() {
        return usersList;
    }

    public long getTotalUsers() {
        return totalUsers;
    }

}

File: app/src/main/java/com/ternaryop/tumblr/TumblrFollowers.kt
package com.ternaryop.tumblr

import org.json.JSONException
import org.json.JSONObject
import java.io.Serializable

/**
 * Created by dave on 18/01/15.
 * Info about followers
 */
@Suppress("MemberVisibilityCanBePrivate")
class TumblrFollowers : Serializable {
    var totalUsers: Long = 0
        private set
    val usersList = mutableListOf<TumblrUser>()

    @Throws(JSONException::class)
    fun add(json: JSONObject) {
        totalUsers = json.getLong("total_users")
        val users = json.getJSONArray("users")
        for (i in 0 until users.length()) {
            usersList.add(TumblrUser(users.getJSONObject(i)))
        }
    }
}
